### PR TITLE
📚 Bookmark Frontend Connection

### LINKED ISSUE #305: Implement Bookmark Frontend Functionality
#### Description
- [x] Implement unfavorite functionality
- [x] Make button "Add Mentors+" lead to gallery
- [x] Connect to backend and update `api.js`

### PR DESCRIPTION
**Pull request by Leundai**
## Description :star2:

Fixes #305 
- Added unfavorite functions
- created endpoint that provided necessary data for bookmark sidebar 

## Screenshots
image
image

### CODE REVIEW

File: backend/api/views/mentee.py [hunk 1]
```diff
@@ -38,3 +38,37 @@ def edit_fav_mentor():
...
+        msg = f"Failed to fetch mentee by id: {id}"
+        logger.info(msg)
+        return create_response(status=422, message=msg)
+
+    favorites = mentee.favorite_mentors_ids
+    logger.info(favorites)
+    mentor_list = list()
+    for mentor_id in favorites:
+        try:
+            mentor = MentorProfile.objects.get(id=mentor_id)
```
Comment (kenetec) [inline backend/api/views/mentee.py:57]: You can use `.only()` to query only the fields that you need to add to the `mentor_list`. Check out [here](https://docs.mongoengine.org/guide/querying.html#retrieving-a-subset-of-fields) for more info

File: backend/api/views/mentee.py [hunk 2]
```diff
@@ -38,3 +38,37 @@ def edit_fav_mentor():
...
+        mentor_list.append(
+            {
+                "name": mentor.name,
+                "title": mentor.professional_title,
+                "id": mentor_id,
+                "image": mentor.image,
+            }
+        )
+
+    logger.info(mentor_list)
```
Comment (kenetec) [inline backend/api/views/mentee.py:70]: Remove the `logger.info` calls in this endpoint

File: frontend/src/utils/api.js [hunk 3]
```diff
@@ -256,7 +256,7 @@ export const deleteMentorById = (id) => {
 export const EditFavMentorById = (mentee_id, mentor_id, favorite) => {
   const requestExtension = `/mentee/editFavMentor`;
   const data = {
-    mentee_uid: mentee_id,
+    mentee_id: mentee_id,
     mentor_id: mentor_id,
     favorite: favorite,
   };
```
Comment (kenetec) [inline frontend/src/utils/api.js:262]: You can actually just pass in the names of the favorites if they are the same as the key ex: 
```
const data = {mentee_id, mentor_id, favorite}
```

File: frontend/src/components/pages/MenteeAppointments.js [hunk 4]
```diff
@@ -47,28 +51,45 @@ function AppointmentCard({ info }) {
...
   const handleOverlayChange = (newSelect) => {
     setVisibleAppts(appointments[newSelect]);
   };
 
+  const handleUnfavorite = async (mentorId, name) => {
+    const res = await EditFavMentorById(profileId, mentorId, false);
+    if (!res) {
+      message.error(`failed to unfavorite mentor ${name}`, 3);
+    } else {
+      message.success(`successfully unfavorite mentor ${name}`, 3);
```
Comment (kenetec) [inline frontend/src/components/pages/MenteeAppointments.js:87]: Extremely minor but you should capitalize the statement.

**Review by kenetec — APPROVED** [review]
amazing work!! 🤩

**Review by AngelaLuo49021 — APPROVED** [review]
Beautful functionality as usual Leo

File: backend/api/views/mentee.py [hunk 5]
```diff
@@ -12,16 +12,17 @@
 def edit_fav_mentor():
     try:
         data = request.get_json()
+        logger.info(data)
```
Comment (Leundai) [inline backend/api/views/mentee.py:15]: wawaweewa

**Pull request merged**